File: AttendanceSystemIPCamera/Models/ChangeRequest.cs
﻿using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace AttendanceSystemIPCamera.Models
{
    public enum ChangeRequestStatus
    {
        UNRESOLVED = 0,
        APPROVED = 1,
        REJECTED = 2
    }
    public class ChangeRequest: BaseEntity
    {
        [Key]
        public int Id { get; set; }
        public Record Record { get; set; }
        public string Comment { get; set; }
        public ChangeRequestStatus Status { get; set; } = ChangeRequestStatus.UNRESOLVED;
        [NotMapped]
        public bool IsResolved => Status != ChangeRequestStatus.UNRESOLVED;
        [JsonIgnore]
        public int RecordId { get; set; }
    }
}

File: AttendanceSystemIPCamera/ClientApp/src/models/ChangeRequest.ts
﻿export enum ChangeRequestStatus {
    UNRESOLVED = 0,
    APPROVED = 1,
    REJECTED = 2
};

export enum ChangeRequestStatusFilter {
    UNRESOLVED = 0,
    RESOLVED = 1,
    ALL = 2
}

export default interface ChangeRequest {
    id: number;
    recordId: number;
    attendeeCode: string;
    attendeeName: string;
    groupName: string;
    groupCode: string;
    sessionId: number;
    sessionTime: Date;
    sessionName: string;
    comment: string;
    status: ChangeRequestStatus;
};
